taichi/codegen/codegen_llvm.cpp

```diff
@@ -1422,6 +1429,7 @@ void CodeGenLLVM::create_offload_struct_for(OffloadedStmt *stmt, bool spmd) {
     if (stmt->bls_epilogue) {
       call("block_barrier");  // "__syncthreads()"
       stmt->bls_epilogue->accept(this);
+      call("block_barrier");  // "__syncthreads()"
```

Comment:
It's unclear to me why we need `__syncthreads()` here, but removing them can lead to some unexpected behavior in simulations.